### ISSUE TITLE
WinHttpHandlerTest.SendAsync_SlowPostRequestWithTimedCancellation_ExpectTaskCanceledException failed in CI

### ISSUE THREAD

**Issue by stephentoub** [dotnet/corefx#3312]
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_release/1240/console
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_windows_release/1245/console
```
System.Net.Http.WinHttpHandlerUnitTests.WinHttpHandlerTest.SendAsync_SlowPostRequestWithTimedCancellation_ExpectTaskCanceledException [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   3
        Stack Trace:
           d:\j\workspace\dotnet_corefx_windows_release\src\System.Net.Http.WinHttpHandler\tests\UnitTests\WinHttpHandlerTest.cs(37,0): at System.Net.Http.WinHttpHandlerUnitTests.WinHttpHandlerTest.Dispose()
              at ReflectionAbstractionExtensions.DisposeTestClass(ITest test, Object testClass, IMessageBus messageBus, ExecutionTimer timer, CancellationTokenSource cancellationTokenSource)
```

**Comment by davidsh** [comment]
I've discovered that the FakeSafeWinHttpHandle leak detection isn't stable.  SafeHandles finalize via the CriticalFinalizer thread.  This thread doesn't seem to run deterministically, so using code like this to wait for SafeHandles to be finalized (and then testing for leaks) doesn't work reliably.

```c#
public static void ForceGarbageCollection()
{
    // Make several passes through the FReachable list.
   for (int i = 0; i < 10; i++)
   {
       GC.Collect();
       GC.WaitForPendingFinalizers();
   }
}
~``

I'm working on a larger PR that has removed this leak detection and I'll be doing leak detection a different way.  I was going to wait until I was finished with that larger PR but if this problem keeps occurring regularly in CI, then I might just disable it now.

**Comment by stephentoub** [comment]
Thanks, David.  Sounds good.  Let's go ahead and disable it in the meantime.  It seems to be taking out a fair number of the release CI runs.

**Comment by davidsh** [comment]
Fixed w/ #3323.
